docs/references/global-shell.md

```diff
@@ -0,0 +1,153 @@
+---
+id: global-shell
+title: Global Shell
+---
+
+The Global App Shell is a new feature in DHIS2 v42. It’s a shared launching point for all of the apps: when you visit DHIS2, the Global Shell will open, and then the Global Shell will load apps inside of itself.
+
+Its intention is to create a consistent experience across DHIS2, consolidate functionality that’s needed across all the apps, and make it easier to roll out shared features to all the apps at once. The first examples of these new features include a redesign of the DHIS2 header bar and the introduction of a new navigation menu, called the Command Palette.
+
+![The Command Palette](./assets/global-shell-command-palette.png)
+
+### Incremental design
+
+The Global Shell is currently designed to require as few changes from apps as possible. In later stages of its implementation, it will require some more updates from apps, but will hoist a number of features that are common to all apps up to itself, improving performance and simplifying the apps. These features might include fetching common requests, managing alerts and localization, and a more semantic connection for routing and PWA communication.
+
+For the future, we’re investigating using the Global Shell to run apps in a sandbox, so that the Global Shell can mediate per-app permissions and improve the security and trustability of third-party apps.
+
+### Table of contents
+
+-   [Components](#components)
+-   [How the routing works](#how-the-routing-works)
+    -   [A new URL scheme](#a-new-url-scheme)
+    -   [Example](#example)
+    -   [Client-side routing and app routing](#client-side-routing-and-app-routing)
+    -   [Bypasses](#bypasses)
+-   [Other features](#other-features)
+    -   [PWA Updates](#pwa-updates)
+    -   [Command palette and key bindings](#command-palette-and-key-bindings)
+-   [Adapting your app to the Global Shell](#adapting-your-app-to-the-global-shell)
+    -   [Plugins and targeting the right `window`](#plugins-and-targeting-the-right-window)
+    -   [Header bars](#header-bars)
+    -   [Hash routing and notifying the Global Shell for deep linking](#hash-routing-and-notifying-the-global-shell-for-deep-linking)
+    -   [External links](#external-links)
+-   [Troubleshooting](#troubleshooting)
+
+## Components
+
+The Global Shell is enabled by the following new components:
+
+1. The Global Shell app, which is hosted at `/apps/`. The app features the following components:
+    1. The header bar, which has gotten a new design
+    2. The Command Palette, a new navigation menu which can be launched from the header bar, but also with a keyboard shortcut
```

Comment:
maybe where there is user-facing doc for command palette, we can link to it here

Reply:
Yeah definitely 🙂